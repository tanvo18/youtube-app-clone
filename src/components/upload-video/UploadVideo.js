import React, { Component } from 'react';
import UploadVideoStyled from './UploadVideoStyled';
import Indicator from '../indicator/Indicator';
import PropTypes from 'prop-types';

class UploadVideo extends Component {
  constructor(props) {
    super(props);
    this.selectedFile = null;
  }

  handleChangeFile = (files) => {
    this.selectedFile = files[0];
    this.props.postVideo(this.selectedFile);
  }

  render() {
    const { isPosting } = this.props;
    
    return (

      <UploadVideoStyled className="upload-btn-wrapper">
        {
          isPosting
            ? <Indicator />
            : <div>
              <input
                className="select-file"
                type="file"
                onChange={(event) => this.handleChangeFile(event.target.files)}
              />
              <button
                className="upload-btn"
              >
                Upload
              </button>
            </div>
        }

      </UploadVideoStyled>
    );
  }
}

UploadVideo.defaultProps = {
  isPosting: false,
  postVideo: () => { },
};

UploadVideo.propTypes = {
  isPosting: PropTypes.bool,
  postVideo: PropTypes.func,
};

export default UploadVideo;