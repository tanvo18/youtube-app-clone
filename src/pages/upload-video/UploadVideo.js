import React, { Component } from 'react';
import UploadVideoContainer from '../../containers/upload-video/UploadVideoContainer';

class UploadVideo extends Component {

  render() {
    return (
      <div>
        <UploadVideoContainer />
      </div>
    );
  }
}

export default UploadVideo;