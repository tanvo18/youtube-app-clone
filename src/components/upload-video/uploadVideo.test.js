import React, { Component } from 'react';
import UploadVideo from './UploadVideo';
import renderer from 'react-test-renderer';

describe('test UploadVideo', () => {
  const postVideo = jest.fn();


  it('test upload button exists', () => {
    const wrapper = shallow(<UploadVideo
      isPosting={false}
      postVideo={postVideo} />);
    expect(wrapper.find('.upload-btn').exists()).toBe(true);
  });

  it('test loading when upload', () => {
    const wrapper = shallow(<UploadVideo
      isPosting={true}
      postVideo={postVideo} />);
    
    expect(wrapper.find('Indicator').exists()).toBe(true);
    expect(wrapper.find('input').exists()).toBe(false);
  });

  it('should render input when upload finish', () => {
    const wrapper = shallow(<UploadVideo
      isPosting={false}
      postVideo={postVideo} />);
    
    expect(wrapper.find('Indicator').exists()).toBe(false);
    expect(wrapper.find('input').exists()).toBe(true);
  });

  it('test function when change file in input', () => {
    const wrapper = shallow(<UploadVideo
      isPosting={false}
      postVideo={postVideo} />);
    const fileContents = 'file contents';
    const file = new Blob([fileContents], { type: 'text/plain' });
    const fileInput = wrapper.find('input');
    fileInput.simulate('change', { target: { files: file } });
    expect(postVideo).toHaveBeenCalled();
  });

  it('should render correctly', () => {
    const tree = renderer
      .create(<UploadVideo
        isPosting={false}
        postVideo={postVideo} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});