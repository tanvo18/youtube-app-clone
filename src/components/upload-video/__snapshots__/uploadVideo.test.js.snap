// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test UploadVideo should render correctly 1`] = `
<div
  className="upload-btn-wrapper sc-bdVaJa kybblu"
>
  <div>
    <input
      className="select-file"
      onChange={[Function]}
      type="file"
    />
    <button
      className="upload-btn"
    >
      Upload
    </button>
  </div>
</div>
`;
